javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/SpanLoggingCustomizerProvider.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfigurationCustomizer;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.ConsoleExporterModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfigurationModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SimpleSpanProcessorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanExporterModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanProcessorModel;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.TracerProviderModel;
+
+/** Adds span logging exporter for debug mode */
+@AutoService(DeclarativeConfigurationCustomizerProvider.class)
+public class SpanLoggingCustomizerProvider implements DeclarativeConfigurationCustomizerProvider {
+
+  @Override
+  public void customize(DeclarativeConfigurationCustomizer customizer) {
+    customizer.addModelCustomizer(
+        model -> {
+          maybeEnableLoggingExporter(model);
+          return model;
+        });
+  }
+
+  private static void maybeEnableLoggingExporter(OpenTelemetryConfigurationModel model) {
+    // read from system properties as it's an early init property and the config bridge is not
+    // available here
+    if (!"true".equals(System.getProperty("otel.javaagent.debug"))) {
+      return;
+    }
```

Comment:
why are you using only system properties?

Reply:
fixed